### ISSUE TITLE
Add test for multiple binders in case expression

### ISSUE THREAD

**Issue by turlando** [turlando/purescm#16]
Compiling this

```
foo 0 0 = 0
foo 0 1 = 1
foo 1 0 = 10
foo _ 2 = 2
foo 3 _ = 30
foo _ _ = 50
```

will produce

```
(define foo
  (lambda (v)
    (lambda (v1)
      (cond
        ((= v 0) 0)
        ((= v1 0) 0)
        ((= v 0) 1)
        ((= v1 1) 1)
        ((= v 1) 10)
        ((= v1 0) 10)
        (? 2)
        ((= v1 2) 2)
        ((= v 3) 30)
        (? 30)
        (? 50)
        (? 50)))))
```

which is plain wrong.

**Comment by f-f** [comment]
I'll hijack this issue to remind us that there should be a test about this

**Comment by f-f** [comment]
Fixed in #18
